### ISSUE TITLE
@typescript-eslint/rule-tester: Support for custom cwd in test configuration

### ISSUE THREAD

**Issue by conarti** [typescript-eslint/typescript-eslint#11668]
## Description

`@typescript-eslint/rule-tester` does not provide a way to set custom `cwd` for the linter used in tests. This makes it impossible to test rules that depend on path resolution with custom working directories or alias imports (like `@/`).

## Expected Behavior

The `RuleTesterConfig` should support a `cwd` option that would be passed to the `Linter` constructor, similar to how it works in ESLint core.

```typescript
const ruleTester = new RuleTester({
  languageOptions: { /* ... */ },
  cwd: '/custom/working/directory',
});
```

## Current Behavior

- There is no `cwd` option in `RuleTesterConfig`
- The linter is created internally with a computed `basePath` based on `tsconfigRootDir` and filename
- Custom working directories cannot be set for test scenarios

## Use Case

We have ESLint rules that need to:
1. Resolve alias imports (e.g., `@/` mapped to `src/`)
2. Work with specific project structures that depend on `context.getCwd()`
3. Test behavior with different working directories

## Workaround

Currently, we have to skip tests that require custom `cwd`:

```typescript
describe.skip('tests with custom cwd', () => {
  // Tests that need custom cwd
});
```

## Implementation Suggestion

The `Linter` is created in `#getLinterForFilename` method. The `cwd` could be:
1. Added to `RuleTesterConfig` interface
2. Passed to the `Linter` constructor instead of the computed `basePath` when explicitly set

## Environment

- `@typescript-eslint/rule-tester`: ^8.45.0
- `eslint`: ^9.11.1
- Node.js: v22.18.0

Would appreciate any guidance or if there's an existing way to achieve this that I'm missing!

**Comment by kirkwaiblinger** [comment]
Closely related to https://github.com/typescript-eslint/typescript-eslint/issues/11274

**Comment by JoshuaKGoldberg** [comment]
`@typescript-eslint/rule-tester` is based off of the core [ESLint `RuleTester`](https://eslint.org/docs/latest/integrate/nodejs-api#ruletester) - which does not have a `cwd`. We'd normally suggest filing the issue to ESLint itself.

For reference, the tseslint class defaults to the `tsconfigRootDir` or `process.cwd()`:

https://github.com/typescript-eslint/typescript-eslint/blob/c7119b5e0fb611bdc55e7caee29e1765b636479a/packages/rule-tester/src/RuleTester.ts#L202-L203

https://github.com/typescript-eslint/typescript-eslint/blob/c7119b5e0fb611bdc55e7caee29e1765b636479a/packages/rule-tester/src/RuleTester.ts#L216

https://github.com/typescript-eslint/typescript-eslint/blob/c7119b5e0fb611bdc55e7caee29e1765b636479a/packages/rule-tester/src/RuleTester.ts#L222-L225

@conarti I would suggest posting a reproduction (https://antfu.me/posts/why-reproductions-are-required)? If `tsconfigRootDir` isn't enough for you it'd be good to know why.

---

@typescript-eslint/triage-team I think this is blocked on discussing adding `cwd` to ESLint core. What do you think?

**Comment by JoshuaKGoldberg** [comment]
The 👍  and another look-through make me more confident here that:

* Most of the issues around the default / implicit `cwd` with `RuleTester` right now are likely due to #11274 
* Otherwise adding an explicit `cwd` option to `RuleTester` would need to be proposed to ESLint first: https://github.com/eslint/eslint/issues

Closing this issue out in the meantime. If and when ESLint core adds a `cwd` to its `RuleTester`, we can re-open or file a new issue to add a corresponding option to `@typescript-eslint/rule-tester`'s. Thanks for filing!
